readers/heuristics/src/test/java/com/criteo/hadoop/garmadon/heuristics/HeuristicsResultDBTest.java

```diff
@@ -0,0 +1,160 @@
+            // Config Embedded Mysql
+            config = aMysqldConfig(v5_7_latest)
+                .withCharset(UTF8)
+                .withPort(port)
+                .withUser(elephant, elephant)
+                .build();
+
+            s.close();
+            initDb();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        dbConfiguration = new DbConfiguration();
+        dbConfiguration.setUser(elephant);
+        dbConfiguration.setPassword(elephant);
+        dbConfiguration.setConnectionString("jdbc:mysql://localhost:" + port + "/" + elephant);
+    }
+
+    @After
+    public void setDown() {
+        mysqld.stop();
+    }
+
+    @Test
+    public void createHeuristicResult() throws SQLException {
+        String appId = "appId";
+        String detail = "detail";
+        HeuristicsResultDB db = new HeuristicsResultDB(dbConfiguration);
+        HeuristicResult heuristicResult = new HeuristicResult(appId, "attemptId", GCCause.class, 1, 1);
+        heuristicResult.addDetail(detail, "value", "This is a new detail");
+        db.createHeuristicResult(heuristicResult);
+
+        ResultSet rset = db.connection.createStatement().executeQuery("SELECT  * FROM garmadon_yarn_app_heuristic_result");
+        HashMap<String, String> result = new HashMap<>();
+        while (rset.next()) {
+            result.put(rset.getString(1), rset.getString(2));
+        }
+        assertEquals(appId, result.get("1"));
+
+        rset = db.connection.createStatement().executeQuery("SELECT  * FROM garmadon_yarn_app_heuristic_result_details");
+        result = new HashMap<>();
+        while (rset.next()) {
+            result.put(rset.getString(1), rset.getString(2));
+        }
+        assertEquals(detail, result.get("1"));
+    }
+
+    @Test
+    public void updateHeuristicHelp() throws SQLException {
+        String help = "<p>\nTest\n</p>";
+        HeuristicsResultDB db = new HeuristicsResultDB(dbConfiguration);
+
+        List<Heuristic> heuristics = new ArrayList<>();
+        Heuristic heuristic = mock(Heuristic.class);
+        doReturn(help).when(heuristic).getHelp();
+        heuristics.add(heuristic);
+        db.updateHeuristicHelp(heuristics);
+
+        ResultSet rset = db.connection.createStatement().executeQuery("SELECT  * FROM garmadon_heuristic_help");
+        HashMap<String, String> result = new HashMap<>();
+        while (rset.next()) {
+            result.put(rset.getString(1), rset.getString(2));
+        }
+        assertEquals(help, result.get(heuristic.getClass().getSimpleName()));
+    }
+
+    @Test
+    public void reconnectOnFailure() {
+        String appId = "appId";
+        String detail = "detail";
+        HeuristicsResultDB db = spy(new HeuristicsResultDB(dbConfiguration));
+        HeuristicResult heuristicResult = new HeuristicResult(appId, "attemptId", GCCause.class, 1, 1);
+        heuristicResult.addDetail(detail, "value", "This is a new detail");
+        new Thread(() -> {
```

Comment:
why stopping in another thread if we waitUntilDbStopped ? There is something I don't get

Reply:
True moving the stop out of th thread and keep the start in it